packages/react/src/useActionForm.ts

```diff
@@ -0,0 +1,293 @@
+  if (modelManager && typeof findBy === "string") {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    return useFindOne(modelManager as any, findBy, options);
+  } else if (modelManager) {
+    const findByKey = Object.keys(findBy)[0];
+    const findByValue = Object.values(findBy)[0];
+    const finder = (modelManager as Record<string, any>)[`findBy${camelize(findByKey)}`];
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    return (useFindBy as any)(finder, findByValue, options);
+  } else {
+    return [
+      { fetching: false },
+      () => {
+        // noop
+      },
+    ];
+  }
+};
+
+const OmittedKeys = ["__typename", "id", "createdAt", "updatedAt"] as const;
+type OmittedKey = (typeof OmittedKeys)[number];
+
+const omitKeys = (data: any) => {
+  const newData = { ...data };
+  for (const key of OmittedKeys) {
+    delete newData[key];
+  }
+  return newData;
+};
+
+const toDefaultValues = (modelApiIdentifier: string | undefined, data: any) => {
+  data = omitKeys(data);
+
+  if (modelApiIdentifier && data[modelApiIdentifier]) {
+    data[modelApiIdentifier] = omitKeys(data[modelApiIdentifier]);
+  }
+
+  return data;
+};
+
+/**
+ * The identity of a record to build a form for
+ *
+ * - Pass the ID of a record as a string to look up a record by id
+ * - Pass a {<fieldValue>: <value>} object to look up a record by some other field. __Note__: a `findBy` function must exist for the field you wish to find the record by.
+ *
+ * @example
+ * // find the user record with ID=123
+ * const { submit } = useActionForm(api.user.update, { record: "123" });
+ *
+ * @example
+ * // find a record by email. `api.user.findByEmail` must exist
+ * const { submit } = useActionForm(api.user.update, { record: { email: "user@gadget.app" } });
+ */
+export type RecordIdentifier = string | { [key: string]: any };
+
+export type UseActionFormResult<
+  GivenOptions extends OptionsType,
+  SchemaT,
+  ActionFunc extends ActionFunction<GivenOptions, any, any, SchemaT, any> | GlobalActionFunction<any>,
+  FormVariables extends FieldValues,
+  FormContext = any
+> = Omit<UseFormReturn<FormVariables, FormContext>, "handleSubmit"> & {
+  /**
+   * Any error that occurred during initial data fetching or action submission
+   */
+  error?: ErrorWrapper | Error | null;
+  /**
+   * Function to call to submit the form
+   */
+  submit: () => Promise<void>;
```

Comment:
If submit is gonna return stuff, can we type it here? 

I think returning stuff is important so that folks can naturally do an `const record = await submit(); redirectTo(record.id` sort of thing

Reply:
whoops yup!

Reply:
in the docs it looks like you expected it to return `result.data` from `useAction`? Or should we return the same as the useAction mutation?

Reply:
Hm, I guess being consistent would be best, I am down for returning the whole result instead of just the data

Reply:
:+1: putting me through TS paces today; I've typed these